connectivity/http/http.c

```diff
@@ -327,16 +331,41 @@ static int request_log(void *cls, const struct sockaddr *addr, socklen_t addrlen
   return MHD_YES;
 }
 
+static status_t build_request(ta_http_request_t *req, const char *data, size_t size) {
+  if (size > 0) {
```

Comment:
what if size == 0 ?

Reply:
I will add codes to check whether `req`, `data` and `size` are legal.